controllers/runner_controller.go

```diff
@@ -927,3 +944,82 @@ func workVolumeMountPresent(items []corev1.VolumeMount) (bool, int) {
 	}
 	return false, 0
 }
+
+func applyWorkVolumeClaimTemplate(runner *v1alpha1.Runner) error {
+	isPresent, _ := workVolumeMountPresent(runner.Spec.VolumeMounts)
+	if isPresent {
+		return errors.New("volumeMount with name \"work\" should not exist if workVolumeClaimTemplate is specified")
+	}
+
+	isPresent, _ = workVolumePresent(runner.Spec.Volumes)
+	if isPresent {
+		return errors.New("volume with name \"work\" should not exist if workVolumeClaimTemplate is specified")
```

Comment:
Open to suggestions regarding error messages and if errors should be of specific type